paddle/fluid/platform/enforce.h

```diff
@@ -125,6 +125,22 @@ struct EOFException : public std::exception {
   const char* what() const noexcept override { return err_str_.c_str(); }
 };
 
+struct NANINFException : public std::exception {
```

Comment:
Why do you add `NANINFException ` here?

Reply:
Because I need to distinguish NAN/INF exception from the others during fast debugging, in order to:

1. handle some exceptions like (summing an empty var) in fast debugging.

2. post-precess (like saving the parameters and persistable vars).